packages/big-design/src/components/List/List.tsx

```diff
@@ -1,34 +1,246 @@
 import { State } from '@popperjs/core';
-import React, { forwardRef, HTMLAttributes, memo, Ref } from 'react';
+import { UseSelectPropGetters } from 'downshift';
+import React, { forwardRef, Fragment, HTMLAttributes, memo, Ref, useCallback, useMemo, useRef } from 'react';
 
 import { useIsomorphicLayoutEffect, useWindowSize } from '../../hooks';
+import { typedMemo } from '../../utils';
+import { Box } from '../Box';
+import { DropdownItem, DropdownItemGroup, DropdownLinkItem, DropdownProps } from '../Dropdown';
+import { SelectAction, SelectOption, SelectOptionGroup, SelectProps } from '../Select';
 
+import { ListGroupHeader } from './GroupHeader';
+import { ListGroupSeparator } from './GroupSeparator';
+import { ListItem } from './Item';
 import { StyledList } from './styled';
 
-export interface ListProps extends HTMLAttributes<HTMLUListElement> {
+export interface ListProps<T> extends HTMLAttributes<HTMLUListElement> {
+  action?: SelectAction;
+  autoWidth: boolean;
+  filteredItems?: Array<SelectOption<T> | SelectAction>;
+  highlightedIndex: number;
   isOpen: boolean;
-  maxHeight: number;
-  update(): Promise<Partial<State> | null>;
+  items: DropdownProps['items'] | SelectProps<T>['options'];
+  isDropdown?: boolean;
+  maxHeight?: number;
+  selectedItem?: SelectOption<T> | null;
+  selectedItems?: Array<SelectOption<T>> | null;
+  addItem?(item: SelectOption<T>): void;
+  getItemProps: UseSelectPropGetters<any>['getItemProps'];
+  getMenuProps: UseSelectPropGetters<any>['getMenuProps'];
```

Comment:
```suggestion
  getItemProps: UseSelectPropGetters<DropdownProps['items'] | SelectProps<T>['options']>['getItemProps'];
  getMenuProps: UseSelectPropGetters<DropdownProps['items'] | SelectProps<T>['options']>['getMenuProps'];
```

Reply:
This is the only change that isn't working. I can't set directly from the Dropdown or Select props because those items contain groups, which Downshift doesn't get. But even setting a type like `DropdownItem | DropdownLinkItem | SelectOption<T> | SelectAction` returns an error on all components, because `Select` (for example) only passes in `SelectOption<T> | SelectAction`.

Reply:
I tried to play around with this for a bit, and yeah it doesn't seem easily possible without separating the `List` component like we've chatted about. Should be good for now.